src/engagement_db_to_analysis/code_imputation_functions.py

```diff
@@ -0,0 +1,81 @@
+from core_data_modules.util import TimeUtils
+
+from core_data_modules.data_models.code_scheme import CodeTypes
+from core_data_modules.cleaners.cleaning_utils import CleaningUtils
+from core_data_modules.traced_data import Metadata
+from core_data_modules.logging import Logger
+
+
+log = Logger(__name__)
+
+def _impute_age_category(user, messages_traced_data, analysis_dataset_configs):
+    """
+    Imputes age category for age dataset messages.
+
+    :param user: Identifier of user running the pipeline.
+    :type user: str
+    :param messages_traced_data: Messages TracedData objects to impute age_category.
+    :type messages_traced_data: list of TracedData
+    :param analysis_dataset_configs: Analysis dataset configuration in pipeline configuration module.
+    :type analysis_dataset_configs: pipeline_config.analysis_configs.dataset_configurations
+    """
+
+    # Get the coding configurations for age and age_category analysis datasets
+    age_category_cc = None
+    for analysis_dataset_config in analysis_dataset_configs:
+        for coding_config in analysis_dataset_config.coding_configs:
+            if coding_config.age_category_config is not None:
+                age_category_cc = coding_config
+
+    age_cc = None
+    age_engagement_db_datasets = None
+    for analysis_dataset_config in analysis_dataset_configs:
+        for coding_config in analysis_dataset_config.coding_configs:
+            if coding_config.analysis_dataset == age_category_cc.age_category_config.age_analysis_dataset:
```

Comment:
again, what happens if there were multiple matches. I think we need to check here (and/or pre-validate when the config gets loaded)